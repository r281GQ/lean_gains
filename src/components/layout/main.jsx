import React from 'react';
import { IconButton, ToolbarGroup, ToolbarTitle } from 'material-ui';
import Menu from 'material-ui/svg-icons/navigation/menu';

const Main = ({ openSideBar }) =>
  <ToolbarGroup firstChild={true}>
    <IconButton tooltip="Main menu" onTouchTap={openSideBar}>
      <Menu color="white" />
    </IconButton>
  </ToolbarGroup>;

export default Main;
