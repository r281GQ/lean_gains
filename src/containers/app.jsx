import React from 'react';
import { Route, Redirect } from 'react-router-dom';

import LoginContainer from './login';
import SignUpContainer from './signup';
import MainContainer from './main';
import withAuthentication from './authentication';

//TODO guard every child route instead of here
const App = () =>
  <div>
    <Route path="/app" component={withAuthentication(MainContainer)} />
    <Route exact path="/login" component={LoginContainer} />
    <Route exact path="/signup" component={SignUpContainer} />
    <Redirect to="/app" />
  </div>;

export default App;
