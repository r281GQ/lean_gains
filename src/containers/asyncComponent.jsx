import React from 'react';

export default loader => (
  class AsyncComponent extends React.Component {
    constructor(props) {
      super(props);

      this.Component = null;
      this.state = { Component: AsyncComponent.Component };
    }

    componentWillMount() {
      if (!this.state.Component) {
        loader().then((Component) => {
          AsyncComponent.Component = Component;

          this.setState({ Component });
        });
      }
    }

    render() {
      if (this.state.Component) {
        return (
          <this.state.Component { ...this.props }  />
        )
      }

      return null;
    }
  }
);
