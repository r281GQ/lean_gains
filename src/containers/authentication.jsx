import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

export default function(WrappedComponent) {
  const mapStateToProps = state => {
    return {
      isAuthenticated: state.getIn(['auth', 'authenticated'])
    };
  };

  class AuthenticationContainer extends PureComponent {
    componentWillMount = () =>
      !this.props.isAuthenticated ? this.props.push('/login') : null;

    componentWillReceiveProps = nextState =>
      !this.props.isAuthenticated ? this.props.push('/login') : null;

    render = () => {
      return <WrappedComponent {...this.props} />;
    };
  }

  return connect(mapStateToProps, { push })(AuthenticationContainer);
}
