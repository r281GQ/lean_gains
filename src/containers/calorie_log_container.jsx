import React, { PureComponent } from 'react';
import { FieldArray, reduxForm } from 'redux-form/immutable';
import { connect } from 'react-redux';
import Nutritions from './../components/calorie_tracker/nutritions';
import { initLog } from './../store/actionCreators/calorie_action_creators';

class CalorieLogContainer extends PureComponent {
  componentDidMount() {
    this.props.dispatch(initLog());
  }

  render() {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <FieldArray name="nutritions" component={Nutritions} />
      </form>
    );
  }
}

export default connect()(
  reduxForm({ form: 'calorie-log' })(CalorieLogContainer)
);
