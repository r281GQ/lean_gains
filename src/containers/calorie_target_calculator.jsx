import React, { Component } from 'react';
import { reduxForm, formValueSelector, initialize } from 'redux-form/immutable';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { Map } from 'immutable';

import { createKcalTarget } from './../store/actionCreators/user_details_action_creators';

import calorieTarget from './../store/selectors/calorie_target';

import CalorieTargetResult from './../components/calorie_target_result';
import BMRCalculationSelector from './../components/bmr_calculation_selector';
import BodyFatField from './../components/body_fat_field';
import ActivityLevelSelecor from './../components/activity_level_selector';
import CalorieSplitSelector from './../components/calorie_split_selector';
import CustomCalorieField from './../components/custom_calorie_field';
import ProteinField from './../components/protein_field';
import FatMethodSelector from './../components/fat_mathod_selector';
import FatSelector from './../components/fat_selector';
import CenteredSubmitButton from './../components/centered_submit_button';

class CalorieTargetCalculator extends Component {
  _adjustCaloriePercentage = (props, nextProps) => {
    const { calorieSplit } = nextProps;
    const { change } = props;
    if (calorieSplit === 'slowbulk') {
      change('restDay', -10);
      change('trainingDay', 30);
    }
    if (calorieSplit === 'cut') {
      change('restDay', -30);
      change('trainingDay', 10);
    }
    if (calorieSplit === 'recomp') {
      change('restDay', -20);
      change('trainingDay', 20);
    }
  };

  _adjustFatRatio = (
    {
      change,
      restFatGrams,
      restFatPercentage,
      trainingFatGrams,
      trainingFatPercentage
    },
    { calorieTarget }
  ) => {
    const maxRestGram = calorieTarget.getIn(['max', 'restGram']);
    const maxTrainingGram = calorieTarget.getIn(['max', 'trainingGram']);
    const maxRestPercentage = calorieTarget.getIn(['max', 'restPercentage']);
    const maxTrainingPercentage = calorieTarget.getIn([
      'max',
      'trainingPercentage'
    ]);

    if (restFatGrams > maxRestGram)
      change('restFatGrams', _.floor(maxRestGram));

    if (trainingFatGrams > maxTrainingGram)
      change('trainingFatGrams', _.floor(maxTrainingGram));

    if (restFatPercentage > maxRestPercentage)
      change('restFatPercentage', _.floor(maxRestPercentage));
    if (trainingFatPercentage > maxTrainingPercentage)
      change('trainingFatPercentage', _.floor(maxTrainingPercentage));
  };

  shouldComponentUpdate = (
    { calorieTarget, calorieSplit, restDay, trainingDay, activity, fatMethod },
    nextState
  ) => {
    const {
      change,
      restFatGrams,
      restFatPercentage,
      trainingFatGrams,
      trainingFatPercentage
    } = this.props;

    const maxRestGram = calorieTarget.getIn(['max', 'restGram']);
    const maxTrainingGram = calorieTarget.getIn(['max', 'trainingGram']);
    const maxRestPercentage = calorieTarget.getIn(['max', 'restPercentage']);
    const maxTrainingPercentage = calorieTarget.getIn([
      'max',
      'trainingPercentage'
    ]);

    const update =
      fatMethod !== this.props.fatMethod ||
      activity !== this.props.activity ||
      calorieSplit !== this.props.calorieSplit ||
      restDay !== this.props.restDay ||
      trainingDay !== this.props.trainingDay ||
      typeof restFatGrams === 'undefined' ||
      typeof this.props.restDay === 'undefined' ||
      restFatGrams > maxRestGram ||
      trainingFatGrams > maxTrainingGram ||
      restFatPercentage > maxRestPercentage ||
      trainingFatPercentage > maxTrainingPercentage;
    return update;
  };

  componentDidMount = () => this.props.initializeForm(this.props);

  componentWillReceiveProps = (nextProps, nextState) => {
    this._adjustCaloriePercentage(this.props, nextProps);
    this._adjustFatRatio(this.props, nextProps);
  };

  render = () => {
    const {
      fatMethod,
      calorieSplit,
      handleSubmit,
      bmrCalculationMethod,
      createCalorieTarget,
      calorieTarget,
      restFatGrams,
      restFatPercentage,
      trainingFatGrams,
      trainingFatPercentage
    } = this.props;
    console.log('rendereds');
    return (
      <div>
        <form onSubmit={handleSubmit(() => createCalorieTarget())}>
          <BMRCalculationSelector />
          <BodyFatField bmrCalculationMethod={bmrCalculationMethod} />
          <ActivityLevelSelecor
            normalize={value =>
              typeof value !== Number ? Number.parseFloat(value) : value}
          />
          <CalorieSplitSelector />
          <CustomCalorieField
            calorieSplit={calorieSplit}
            minRest={calorieTarget.toJS().minCalorie}
            minTraining={calorieTarget.toJS().minCalorie}
            normalize={value =>
              typeof value !== Number ? Number.parseFloat(value) : value}
          />
          <ProteinField />
          <FatMethodSelector />
          <FatSelector
            fatMethod={fatMethod}
            maxRestFatGrams={calorieTarget.getIn(['max', 'restGram'])}
            maxRestFatPercentage={calorieTarget.getIn([
              'max',
              'restPercentage'
            ])}
            maxTrainingFatGrams={calorieTarget.getIn(['max', 'trainingGram'])}
            maxTrainingFatPercentage={calorieTarget.getIn([
              'max',
              'trainingPercentage'
            ])}
          />
          <CenteredSubmitButton label="Create calorie target" />
        </form>

      </div>
    );
  };
}

const selector = formValueSelector('calorie-target');

CalorieTargetCalculator = connect(state => ({
  activity: selector(state, 'activity'),
  fatMethod: selector(state, 'fatMethod'),
  calorieSplit: selector(state, 'calorieSplit'),
  bmrCalculationMethod: selector(state, 'bmrCalculationMethod'),
  trainingFatGrams: selector(state, 'trainingFatGrams'),
  restFatGrams: selector(state, 'restFatGrams'),
  trainingFatPercentage: selector(state, 'trainingFatPercentage'),
  restFatPercentage: selector(state, 'restFatPercentage'),
  restDay: selector(state, 'restDay'),
  trainingDay: selector(state, 'trainingDay')
}))(CalorieTargetCalculator);

const mapStateToProps = state => {
  return {
    calorieTarget: calorieTarget(state)
  };
};

const mapDispatchToProps = dispatch => {
  return {
    initializeForm: ({ calorieTarget }) =>
      dispatch(
        initialize(
          'calorie-target',
          Map().withMutations(map =>
            map
              .set('bodyFat', calorieTarget.get('bodyFat'))
              .set('restDay', -20)
              .set('trainingDay', 20)
              .set('restFatGrams', 0)
              .set('trainingFatGrams', 0)
              .set('restFatPercentage', 0)
              .set('trainingFatPercentage', 0)
              .set('fatMethod', 'grams')
              .set('activity', 1.2)
              .set('protein', 2)
              .set('calorieSplit', 'recomp')
              .set('bmrCalculationMethod', 'harris-benedict')
          )
        )
      ),
    createCalorieTarget: calorieTarget =>
      dispatch(createKcalTarget(calorieTarget))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  reduxForm({ form: 'calorie-target' })(CalorieTargetCalculator)
);
