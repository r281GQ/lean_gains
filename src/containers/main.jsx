import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';

import HeaderContainer from './header';
import SideBarContainer from './side_bar';

import UserDetailsContainer from './user_details';
import WorkoutLogsRouter from './workout_logs/workout_logs_router';
import DailyLogsRouter from './daily_logs/daily_logs_router';
import KcalTracker from './kcal_tracker';
import { initFetch } from './../store/actionCreators/user_details_action_creators';
import { closeSideBar } from './../store/actionCreators/app_action_creators';

import WorkoutTargetRouter from './workout_targets/workout_targets_router';

import CalorieTargetContainer from './calorie_target_container';

import withMessageBar from './massage_container';
import withLoadingScreen from './loading_container';
import withDataCheck from './with_data_check';

const Error = ({errors}) => <div>{errors.map(item => <div>{`${item} is missing First you need to provide this`}</div>)}</div>

//TODO implement graphs and charts
//TODO proptypes
class MainContainer extends PureComponent {
  componentDidMount = () => this.props.initFetch();

  render = () =>
    <div>
      <HeaderContainer />
      <SideBarContainer />
      <Route path="/app/userdetails" component={UserDetailsContainer} />
      <Route path="/app/workoutlogs" component={withLoadingScreen(withMessageBar(WorkoutLogsRouter))} />
      <Route path="/app/dailylogs" component={DailyLogsRouter} />
      <Route path="/app/workouttargets" component={WorkoutTargetRouter} />
      <Route path="/app/kcaltarget" component={withDataCheck(CalorieTargetContainer, [{name:'sex', path:['userDetails', 'sex']}],Error )} />
      <Route path="/app/kcaltracker" component={KcalTracker} />
    </div>;
}

const mapStateToProps = state => ({
  isLoading: state.getIn(['auth', 'isLoading'])
});

const mapDispatchToProps = dispatch => ({
  initFetch: () => dispatch(initFetch())
});

export default connect(mapStateToProps, mapDispatchToProps)(MainContainer);
