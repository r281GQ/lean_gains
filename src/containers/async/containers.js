import asyncComponent from './async_container';

export const AsyncMainContainer = asyncComponent(() =>
  System.import('./../main').then(module => module.default)
);

export const AsyncDailyLogsRouter = asyncComponent(() =>
  System.import('./../daily_logs/daily_logs_router').then(
    module => module.default
  )
);

export const AsyncCalorieTargetContainer = asyncComponent(() =>
  System.import('./../calorie_target/calorie_target_container').then(module => module.default)
);

export const AsyncUserDetailsContainer = asyncComponent(() =>
  System.import('./../user_details').then(module => module.default)
);

export const AsyncWorkoutLogsRouter = asyncComponent(() =>
  System.import('./../workout_logs/workout_logs_router').then(
    module => module.default
  )
);

export const AsyncWorkoutTargetsRouter = asyncComponent(() =>
  System.import('./../workout_targets/workout_targets_router').then(
    module => module.default
  )
);

export const AsyncCalorieTrackerContainer = asyncComponent(() =>
  System.import('./../calorie_tracker/calorie_tracker_container').then(module => module.default)
);
