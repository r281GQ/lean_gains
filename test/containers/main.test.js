import React from 'react';
import { shallow } from 'enzyme';
import { PureMainContainer } from './../../src/containers/main';

describe('Main test', () => {
  it('should render', () => {
    const component = shallow(<PureMainContainer />);
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
