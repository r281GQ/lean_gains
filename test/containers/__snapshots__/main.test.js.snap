// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test should render 1`] = `
<div
  className="main-container"
>
  <Connect(Error) />
  <Connect(SideBarContainer) />
  <Route
    component={[Function]}
    path="/app/userdetails"
  />
  <Route
    component={[Function]}
    path="/app/workoutlogs"
  />
  <Route
    component={[Function]}
    path="/app/dailylogs"
  />
  <Route
    component={[Function]}
    path="/app/workouttargets"
  />
  <Route
    component={[Function]}
    path="/app/kcaltarget"
  />
  <Route
    component={[Function]}
    path="/app/kcaltracker"
  />
</div>
`;
